File: src/main/java/com/hedvig/backoffice/graphql/types/account/AccountEntryInput.java
package com.hedvig.backoffice.graphql.types.account;

import com.hedvig.backoffice.services.account.dto.AccountEntryType;
import lombok.Value;

import javax.money.MonetaryAmount;
import java.time.Instant;
import java.time.LocalDate;
import java.util.Optional;

@Value
public class AccountEntryInput {
  AccountEntryType type;
  MonetaryAmount amount;
  LocalDate fromDate;
  String reference;
  String source;
  Optional<String> title;
  Optional<String> comment;
}

File: src/main/java/com/hedvig/backoffice/graphql/types/account/AccountEntryInput.kt
package com.hedvig.backoffice.graphql.types.account

import java.time.LocalDate
import javax.money.MonetaryAmount

data class AccountEntryInput(
  val type: String,
  val amount: MonetaryAmount,
  val fromDate: LocalDate,
  val reference: String,
  val source: String,
  val title: String?,
  val comment: String?
)
